\documentclass[12pt]{article}  % define document class

\usepackage[utf8]{inputenc} % use UTF-8
\usepackage{pdfpages} % package that allows the inclusion of pdf pages
\usepackage[a4paper, total={6in, 9in}]{geometry} % package for A4
\usepackage[english]{babel}
\usepackage[backend=biber]{biblatex}
\usepackage{graphicx}  % package that adds images
\usepackage{indentfirst}
\usepackage{helvet} % load Helvetica package

\addbibresource{bib.bib}

\renewcommand{\familydefault}{\sfdefault} % change font to Helvetica

\renewcommand{\baselinestretch}{1.5}  % change line spacing


\title{BINF assignment 1}
\author{Gil Oliveira}
\date{April 2020}

\begin{document}

\includepdf[pages=-]{cover.pdf}

\begin{abstract}
	Consensus sequences are very common themes in Molecular Biology and Bioinformatics, allowing for evolution and molecular interaction studies, to name a few examples. There's a big need for accurate and intuitive sequence visualisation techniques, as they can portray much more relevant information than simply typing the sequence letters. The Sequence Logo was the first sequence visualisation technique, and is the most used nowadays, but new paradigms, such as the ProfileGrids, have been appearing to try to compensate for the limitations of the Sequence Logo. In this paper, I compare the most relevant sequence visualisation tools, evaluate their upsides and downsides and try to match them with their most appropriate use cases.
\end{abstract}

\newpage

\section{Introduction}

Consensus sequences are sequences of nucleotides (on DNA and RNA) or amino acids (in protein), which are comprised of the most commonly encountered letters at that postion \parencite{pierce2012}. These sequences are generally associated with inter- or intramolecular interactions \parencite{liljas2001}. A prime example is the Shine-Dalgarno (SD) sequence in prokaryotes, which is involved in the binding of the ribossome to the mRNA.

Even though these sequences are highly conserved, they do present some variations in between organisms and thus, mearly writing out the consensus sequence leaves out information about the frequence and variation of each of the nucleotides or aminoacids in the sequence, which can be very important in Molecular Biology or Bioinformatics analysis.

Let's take the example of the aforementioned consensus sequence. In \textit{E. coli} the SD sequence is \textbf{5'-AGGAGG-3'}. This sequence, however, has been shown to have slight variations in different bacteria \parencite{ma2002} and thus a bioinformatician, when programming a gene prediction tool, may be misled to believe that that's the full extent of the SD sequence, when it's not. Moreover, one may argue that studying the variety of consensus sequences is important in undertanding intermollecular interactions.

Moreover, one study \parencite{cook2008} arrived at the notion that graphical evidence, as opposed to evidence with a text baseline, helps intelligence analysis, such as medical and military decisions, by allowing experts to provide a more balanced and less biased decision. Given the fact that these sequences can have biomedical relevance, we can postulate that these more visual representations of data may help decrease bias in interpretation of disease-related sequences, for instance.

We then arrive at the logical conclusion that the visualisation of consensus sequences plays a very important role in various Molecular Biology and Bioinformatics studies and is therefore relevant to study better ways to display the available data in a way that's informative, visually appealing and easy to understand. Before arriving at a consensus sequence, one has to first perform a multiple sequence alignment (MSA). The tools that will be discussed and compared in this work have the ability to visually display MSAs and can be a very powerful tool in consensus sequence analysis.

\section{Sequence Logos}

Faced with the challenges of creating a visual representation of consensus sequences that provides more information than just the mere sequence of letters, Schnider and Stephens came forward with a proposal for a new paradigm in consensus sequences visualisation, for which they called Sequence Logos (Fig. \ref{fig:eColiSequenceLogo})  \parencite{schneider1990}.

\begin{figure}[!b]
    \centering
    \includegraphics[scale=0.4]{sequence_logos}
    \caption{The Sequence Logo representation of the \textit{E. coli} binding site \protect \parencite[p. 2]{schneider1990}.}
    \label{fig:eColiSequenceLogo}
\end{figure}

The Sequence Logos are comprised of four elements: the letters, which are shaped to become the "bar" of the chart; the colour, which is used to differentiate each of the symbols, even though a grayscale can also be employed; the height, which represents the conservation level of the residue at a particular alignment column and the axes, the horizontal one representing the location and the vertical the frequency and conservation level of the symbols \parencite{nungkion2012}.

Making Sequence Logos nowadays is very easy, given the fact that there are free tools available. One such example is WebLogo, that automatically generates Sequence Logos from MSAs \parencite{crooks2004}. 

As we can see, this graphical method of representing data is a major step forward in relation with what existed before, as these logos allowed us to get a better feel for the relative frequencies of the residues at every position, which is crucial for a more complete understanding of such sequences.

It should then come as no surprise that these logos have become ubiquitous in many Molecular Biology and Bioinformatics studies, being the preferred method of visually displaying consensus sequences.

\section{Shortcomings of Sequence Logos}

Of course, like most visual representations of data, Sequence Logos are not without its drawbacks. Many of which are derived from the fact that, when converting raw data to a some form of visual display, there is some information that is lost in that process. There can be instances in which that information can have some relevance in the study, so researchers should beware of relying solely on these logos to study consensus sequences.

\subsection{Biases in Sequence Logo analysis}

Even though graphical tools, in general, have a tendency to minimize biases in decision-making \parencite{cook2008}, one should tread carefully in asserting that Sequence Logos are ideal ways to analyse biological data.

A 2012 study by Nung Kion and Yin Bee \parencite{nungkion2012} has uncovered that researchers have a tendency to misuse Sequence Logos in computational transcription factor analysis. The authors arrived at that conclusion by analysing published articles and studying the validity of the conclusions derived from Sequence Logo analysis. That analysis yielded that, in biological assays, the use of these logos tends to be more accurate because researchers complement their studies with statistical testing and additional biological data, rather than only relying on the logo itself. In sharp contrast, there is a significant tendency for biases in computational analysis, mainly confirmation bias \parencite{nungkion2012}.

These judgement errors usually occurred when researchers compared their algorithm with another by visual comparison of the two Sequence Logos \parencite{nungkion2012}. These comparisons, according to the authors, are unfair because the Sequence Logo does not provide enough information to objectively access the quality difference of the two methods, adding that these depictions were not designed to allow easy comparison of similarity between them without expert knowledge on the motif being studied.

Perhaps one of the main reason why comparing Sequence Logos is so prone to bias is because scientific findings rely on reliable and systematic evidence to be corroborated, whereas evidence that relies solely on visual interpretations, like in these cases, is not solid enough to meet the criteria associated with the scientific method \parencite{nungkion2012}.


\subsection{Readability and usability}

The design of the Web Logo is far from ideal, as the letters are distorted to represent the conservation and frequency of the residue. That means readability takes a bit hit, as it's harder to distinguish the letters if they're very distorted, particularly the ones that have less significance. One could argue that those, because they're less frequent, are less important, but in protein Sequence Logos, that have many more letters than nucleic acid Sequence Logos, important information may be missed due to poor readability, especially if it's a small figure in a printed paper.

On that note, the fact that Sequence Logos are static and non-interactive also limits the amount of information that they can carry and how they involve the reader. These are, as stated by its original authors, simple tools \parencite{schneider1990}, so one can not expect much user intractability from them, as they were designed to be placed on printed materials and to provide the most relevant information at a glance.

Some authors \parencite{nungkion2012} also argue that the use of colours can be a drawback as well, as visualisations may appear more convincing than they really are, the colours in a Sequence Logo may impress the reader more than the actual quality of the motifs presented.


\section{New approaches}

Since the introduction of the Sequence Logo there have been new approaches that attempt to mitigate the shortcomings of Sequence Logos. Many of them have been incremental and use-specific (e.g. WebLogo and HMMLogo) but others, such as ProfileGrids, which will be discussed ahead, have taken bigger strides \parencite{roca2008}.

\subsection{ProfileGrids}

In an attempt to overcome the shortcomings of Sequence Logos, and to allow for the visual representation of large data sets, Roca, Almada and Abaijan introduced, in 2008, a new paradigm in sequence aligment visualisation, for which they called ProfileGrids (Fig. \ref{fig:profileGrids}) \parencite{roca2008}.

\begin{figure}[!b]
    \centering
    \includegraphics[scale=0.5]{profilegrids}
    \caption{ProfileGrid visualisation of the AKL alignment protein residue distributions. Adapted from the original article \protect \parencite[p. 4]{roca2014}.}
    \label{fig:profileGrids}
\end{figure}

In these representations, each residue occupies a different line, and each position a different column, including the gaps in the sequences.

When compared to Sequence Logos, profile grids offer several advantages: 1) fixed height of residue symbols, allowing for clear reading of all symbols; 2) colour shading of cells, representing the relative frequency of each residues \parencite{roca2014}.

This paradigm also encompasses some degree of interactivity though a Java program named JProfileGrid, which combines the examining of aminoacid frequences across an MSA, identification of conserved motif regions and allows for comparing species-specific residues against a sequence family \parencite{roca2008}.

This paradigm builds upon previous the Sequence Logo and its derivations, offering a more suitable and interactive visualisation scheme for MSAs, having been showcased at the 2013 BioVis Redesign Contest. \parencite{roca2014}


\section{Conclusion}

Visual representations of consensus sequences and other MSAs play an important role in giving scientists a better feel for a plethora of aspects surrounding relevant aminoacid and nucleotide motifs and can influence decision-making in related fields.

The more ubiquitous form of sequence visualisation, the Sequence Logo, is a simple approach to visualise motifs, and can be really useful in quickly giving the reader the most conserved residues, but it trades off the ability to convey information about larger data sets and to clearly display aminoacid sequences, being better suited for nucleotide sequences with short sizes.

The spiritual successor to these logos are Profile Grids, which overcome many of the limitations os Sequence Logos, especially for large aminoacid data sets, but are less intuitive and more cluttered, but with much better readability.

The logical conclusion is that visual representations of data should preferably be used in addition to other forms of testing and data analysis and should not be used on their own to compare different assays and that the most appropriate form of visual representation of motifs varies based on factors such as the type of molecule, the size of the data set and the length of the sequences.


\newpage    % start new page
\printbibliography

\end{document}